cli/factory.go

```diff
@@ -30,6 +40,84 @@ var factoryCLICommand = cli.Command{
 	},
 }
 
+type cacheServer struct {
+	rpc     *grpc.Server
+	factory vc.Factory
+}
+
+var jsonVMConfig *pb.GrpcVMConfig
+
+// Config requests base factory config and convert it to gRPC protocol.
+func (s *cacheServer) Config(ctx context.Context, empty *types.Empty) (*pb.GrpcVMConfig, error) {
+	if jsonVMConfig == nil {
+		config := s.factory.Config()
+
+		var err error
+		jsonVMConfig, err = config.ToGrpc()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return jsonVMConfig, nil
+}
+
+// GetBaseVM requests a paused VM and convert it to gRPC protocol.
+func (s *cacheServer) GetBaseVM(ctx context.Context, empty *types.Empty) (*pb.GrpcVM, error) {
+	config := s.factory.Config()
+
+	vm, err := s.factory.GetBaseVM(ctx, config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to GetBaseVM")
+	}
+
+	return vm.ToGrpc(config)
+}
+
+func getUnixListener(path string) (net.Listener, error) {
+	err := os.MkdirAll(filepath.Dir(path), 0755)
+	if err != nil {
+		return nil, err
+	}
+	if err = unix.Unlink(path); err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+	l, err := net.Listen("unix", path)
+	if err != nil {
+		return nil, err
+	}
+	if err = os.Chmod(path, 0660); err != nil {
```

Comment:
Does this need to be `660`? Could we use `600` instead?

Reply:
Updated.

And I opened an issue for needing CI test for vmcache in https://github.com/kata-containers/tests/issues/1259

Reply:
Thanks @teawater !